### PR TITLE
Refactor quitting

### PR DESCRIPTION
**Pull request by gperciva**

### CODE REVIEW

**Comment by gperciva** [comment]
I think there's more work that I can do to separate the functionality, but this is a good first step.  It's certainly worth discussing at this point.

Future plans: ideally I'd like `closeEvent()` to call `displayStopTasksDialog()` directly, and cancel any tasks itself (if desired), but it seems that `task->stop()` doesn't block.  Currently, we just go ahead and quit anyway (in `closeEvent()` when `_aboutToQuit` is true), but I'd feel more comfortable if we made sure the processes were terminated first.  I need to learn more about Qt and `TarsnapTask` first, though.

**Comment by shinnok** [comment]
I'll be inclined to leave this for a subsequent release, since modifying the app quit logic right now would have me take master out of code freeze and require another round of platform testing which I just completed. :-)

We can take this on a v1.1 branch if that is fine with you?

**Comment by cperciva** [comment]
This seems like something which would be good to sort out before we ship a release called "1.0".  We could ship a "0.99" now, I guess?  But it seems premature to ship 1.0 when there are things we know need to get ironed out... I mean, that's why you sent out an email encouraging people to test before the release, right?

File: src/coreapplication.cpp [hunk 1]
```diff
@@ -163,7 +163,7 @@ void CoreApplication::showMainWindow()
     disconnect(&_notification, &Notification::messageClicked, this,
                &CoreApplication::showMainWindow);
 
-    _mainWindow = new MainWindow();
+    _mainWindow = new MainWindow(&_taskManager);
```
Comment (shinnok) [inline src/coreapplication.cpp]: Passing TaskManager to MainWindow like that breaks the entire encapsulation and loose binding design in that:
1. TaskManager doesn't need to know/care about what's on top of it;
2. MainWindow doesn't have to be tightly coupled with TaskManager in any way;
The only information passing method right now between the UI and backend code is strictly Qt signals&slots mechanism. This implies that the connection between the UI and backend components is loose, non-binding, self encapsulated and segregated. If one was to extend his imagination, the current cross-platform backend is easily portable to any other type of interface where Qt is available (mobile, embedded or headless daemon), which was part of the original design. I think it's best that we refrain from tightly coupling backend code with UI if there's no outstanding argument against that.

**Comment by gperciva** [comment]
Updated PR.  I have more changes planned, but this is a step in the right direction.

**Pull request merged**